larrecodnn/ImagePatternAlgs/Triton/Tools/PointIdAlgSonicTriton_tool.cc

```diff
@@ -0,0 +1,191 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Class:       PointIdAlgSonicTriton_tool
+// Authors:     M.Wang
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "art/Utilities/ToolMacros.h"
+#include "larrecodnn/ImagePatternAlgs/ToolInterfaces/IPointIdAlg.h"
+#include "larrecodnn/ImagePatternAlgs/NuSonic/Triton/TritonClient.h"
+
+namespace PointIdAlgTools {
+
+  class PointIdAlgSonicTriton : public IPointIdAlg {
+  public:
+    explicit PointIdAlgSonicTriton(fhicl::Table<Config> const& table);
+
+    std::vector<float> Run(std::vector<std::vector<float>> const& inp2d) const override;
+    std::vector<std::vector<float>> Run(std::vector<std::vector<std::vector<float>>> const& inps,
+                                        int samples = -1) const override;
+
+  private:
+    std::string fTritonModelName;
+    std::string fTritonURL;
+    bool fTritonVerbose;
+    std::string fTritonModelVersion;
+    unsigned fTritonTimeout;
+    unsigned fTritonAllowedTries;
+
+    std::unique_ptr<TritonClient> triton_client;
+  };
+
+  // ------------------------------------------------------
+  PointIdAlgSonicTriton::PointIdAlgSonicTriton(fhicl::Table<Config> const& table)
+    : img::DataProviderAlg(table())
+  {
+    // ... Get common config vars
+    fNNetOutputs = table().NNetOutputs();
+    fPatchSizeW = table().PatchSizeW();
+    fPatchSizeD = table().PatchSizeD();
+    fCurrentWireIdx = 99999;
+    fCurrentScaledDrift = 99999;
+
+    // ... Get "optional" config vars specific to tRTis interface
+    std::string s_cfgvr;
+    bool b_cfgvr;
+    if (table().TritonModelName(s_cfgvr)) { fTritonModelName = s_cfgvr; }
+    else {
+      fTritonModelName = "mycnn";
+    }
+    if (table().TritonURL(s_cfgvr)) { fTritonURL = s_cfgvr; }
+    else {
+      fTritonURL = "localhost:8001";
+    }
+    if (table().TritonVerbose(b_cfgvr)) { fTritonVerbose = b_cfgvr; }
+    else {
+      fTritonVerbose = false;
+    }
+    if (table().TritonModelVersion(s_cfgvr)) { fTritonModelVersion = s_cfgvr; }
+    else {
+      fTritonModelVersion = "";
+    }
+
+    // ... Create parameter set for Triton inference client
+    fhicl::ParameterSet TritonPset;
+    TritonPset.put("serverURL",fTritonURL);
+    TritonPset.put("verbose",fTritonVerbose);
+    TritonPset.put("modelName",fTritonModelName);
+    TritonPset.put("modelVersion",fTritonModelVersion);
+    TritonPset.put("timeout",fTritonTimeout);
+    TritonPset.put("allowedTries",fTritonAllowedTries);
+    TritonPset.put("outputs","[]");
+    
+    // ... Create the Triton inference client
+    triton_client = std::make_unique<TritonClient>(TritonPset);
```

Comment:
Could the interface of`TritonClient` be adjusted so a pointer is not necessary (e.g.):

```c++
PointIdAlgSonicTriton::PointIdAlgSonicTriton(fhicl::Table<Config> const& table)
  : img::DataProviderAlg(table())
  , fTrtitonURL{...}
  , fTritonVerbose{...}
  , ...
  , triton_client{fTritonURL, fTritonVerbose, ...}
{}
```

Not a huge deal, but would be preferable to avoid pointers where possible.
```

Reply:
I think I'll leave this as is for now.